app/models/account/data_enricher.rb

```diff
@@ -8,49 +8,61 @@ def initialize(account)
   end
 
   def run
-    enrich_transactions
-  end
+    total_unenriched = account.entries.account_transactions
+      .where("enriched_at IS NULL OR merchant_id IS NULL OR category_id IS NULL")
+      .count
 
-  private
-    def enrich_transactions
-      candidates = account.entries.account_transactions.includes(entryable: [ :merchant, :category ])
+    if total_unenriched > 0
+      batch_size = 50
+      batches = (total_unenriched.to_f / batch_size).ceil
 
-      Rails.logger.info("Enriching #{candidates.count} transactions for account #{account.id}")
+      batches.times do |batch|
+        EnrichTransactionBatchJob.perform_later(account, batch_size, batch * batch_size)
+      end
+    end
+  end
 
-      merchants = {}
+  def enrich_transaction_batch(batch_size = 50, offset = 0)
+    candidates = account.entries.account_transactions
+      .includes(entryable: [ :merchant, :category ])
+      .where("enriched_at IS NULL OR merchant_id IS NULL OR category_id IS NULL")
+      .offset(offset)
+      .limit(batch_size)
 
-      candidates.each do |entry|
-        if entry.enriched_at.nil? || entry.entryable.merchant_id.nil? || entry.entryable.category_id.nil?
-          begin
-            next unless entry.name.present?
+    Rails.logger.info("Enriching batch of #{candidates.count} transactions for account #{account.id} (offset: #{offset})")
 
-            info = self.class.synth_provider.enrich_transaction(entry.name).info
+    merchants = {}
 
-            next unless info.present?
+    candidates.each do |entry|
+      begin
+        next unless entry.name.present?
```

Comment:
Since this was originally written, we've added null validations to ensure `name` is present, so this can safely be removed now.